etl/steps/data/garden/health/2023-04-18/wgm_mental_health.py

```diff
@@ -0,0 +1,339 @@
+    # Merge
+    df = pd.concat([df_countries, df_world, df_continents, df_incomes], ignore_index=True)
+    return df
+
+
+def _build_df_with_continents(df: pd.DataFrame) -> pd.DataFrame:
+    # Obtain dataframe for continents
+    df_continents = df.copy()
+    continents = ["Africa", "Asia", "Europe", "North America", "Oceania", "South America"]
+    df_continents["country"] = df_continents["country"].cat.add_categories(continents)
+    for continent in continents:
+        countries = geo.list_countries_in_region(continent)
+        msk = df_continents["country"].isin(countries)
+        df_continents.loc[msk, "country"] = continent
+
+    # Sanity check
+    unc_countries = set(df_continents.country).difference(continents)
+    assert not unc_countries, f"Some countries do not belong to any continent: {unc_countries}!"
+    return df_continents
+
+
+def _build_df_with_incomes(df: pd.DataFrame) -> pd.DataFrame:
+    # Obtain dataframe for income groups
+    df_income = df.copy()
+    incomes = [
+        "High-income countries",
+        "Upper-middle-income countries",
+        "Lower-middle-income countries",
+        "Low-income countries",
+    ]
+    df_income["country"] = df_income["country"].cat.add_categories(incomes)
+    for income in incomes:
+        countries = geo.list_countries_in_region(income)
+        msk = df_income["country"].isin(countries)
+        df_income.loc[msk, "country"] = income
+
+    # Sanity check
+    unc_countries_expected = {"Venezuela"}  # WB unclassified Venezuela in 2021, bc of lacking data
+    unc_countries = set(df_income.country).difference(incomes)
+    assert (
+        unc_countries == unc_countries_expected
+    ), f"Only Venezuela is expected to be unclassified, but found {unc_countries} to be unclassified!"
+    # Remove Venezuela
+    df_income = df_income[~df_income["country"].isin(unc_countries_expected)]
+    return df_income
+
+
+def _make_df_with_share_answers(df: pd.DataFrame, weight_column: str = "weight_intra_country") -> pd.DataFrame:
+    # 1. broken down by gender and age group
+    df_gender_age = make_individual_df_with_share_answers(
+        df, dimensions=["gender", "age_group"], weight_column=weight_column
+    )
+    # 2. broken down by gender
+    df_gender = make_individual_df_with_share_answers(df, dimensions=["gender"], weight_column=weight_column)
+    # 3. broken down by age_group
+    df_age = make_individual_df_with_share_answers(df, dimensions=["age_group"], weight_column=weight_column)
+    # 4. no breakdown
+    df_nb = make_individual_df_with_share_answers(df, weight_column=weight_column)
+    # Combine dataframes
+    log.info("wgm_mental_health: combining dataframes into combined one")
+    df_combined = pd.concat([df_nb, df_gender, df_age, df_gender_age], ignore_index=True)
+    # Sanity check
+    x = df_combined.groupby(["country", "year", "question", "gender", "age_group"], observed=True)[["share"]].sum()
+    assert x[
+        x.share - 100 > 0.1
```

Comment:
A safer approach would be to use the absolute deviation, e.g. `abs(x.share - 100) > 0.1`